src/profiling/symbolizer/llvm_symbolizer_c_api.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SRC_PROFILING_SYMBOLIZER_LLVM_SYMBOLIZER_C_API_H_
+#define SRC_PROFILING_SYMBOLIZER_LLVM_SYMBOLIZER_C_API_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define LLVM_SYMBOLIZER_C_API __attribute__((visibility("default")))
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Opaque handle to the underlying C++ symbolizer object.
+struct LlvmSymbolizer;
+typedef struct LlvmSymbolizer LlvmSymbolizer;
+
+// Represents a single symbolization request.
+typedef struct {
+  const char* binary_path;
+  uint64_t address;
+} SymbolizationRequest;
+
+// Represents a single symbolized stack frame.
+typedef struct {
+  const char* function_name;
+  const char* file_name;
+  uint32_t line_number;
+} SymbolizedFrame;
+
+// Represents the result of a single symbolization operation as a range in a
+// flattened array of frames.
+typedef struct {
+  // The offset in the `frames` array of the `BatchSymbolizationResult`.
+  size_t offset;
```

Comment:
Here and below: avoid size_t across ABI boundaries as that can cause very hard to debug issues at it depends on the arch the library was compiled on. Use uint32_t.